### DISCUSSION TITLE
LinkProps doesn't have aria-current

### DISCUSSION THREAD

**Discussion by andrecasal** [adobe/react-spectrum#6419] [Q&A]
### Context
I'm building a [component library](https://verveui.pro/).
I need to integrate React Aria's Link component's funcionality with that of other framework's Link component (like [remix](https://remix.run/docs/en/main/components/link) or [next](https://nextjs.org/docs/pages/api-reference/components/link)).
[Those frameworks don't provide useLink() hooks to separate functionality from structure.](https://github.com/remix-run/react-router/issues/7390)
So, I need to replicate the React Aria's Link component and render it with that framework's <Link>.

### Problem
Going through the [Link.tsx](https://github.dev/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Link.tsx) code, I noticed the `isCurrent: !!props['aria-current'],` line: `props` is typed as `LinkProps`, but `LinkProps` doesn't have an `aria-current` index.

Am I missing something?

**Comment by snowystinger** [comment]
I don't quite follow the question. I think you're asking how `aria-current?: boolean` is getting applied to `LinkProps`?
That is happening here
```
export interface LinkProps extends Omit<AriaLinkOptions, 'elementType'>, HoverEvents, RenderProps<LinkRenderProps>, SlotProps {}
```
specifically from `RenderProps<LinkRenderProps>`

**Reply by snowystinger** [reply]
O, I see, it's not in our docs props table https://react-spectrum.adobe.com/react-aria/Link.html#props
But I can use a link like 
```
    <Link data-testid="link-example" href="https://www.imdb.com/title/tt6348138/" hrefLang="en"  target="_blank">
      The missing link
    </Link>
```
without a typescript error, so it has to be in there somewhere. Might just be an issue with the docs generation.
